### PR TITLE
Add support for the new user avatar API endpoint

### PR DESCRIPTION
**Pull request by boxtcarpel**
Added the BOXUserAvatarRequest to fetch a user's avatar photo.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @boxtcarpel has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+File.h [hunk 1]
```diff
@@ -19,7 +19,6 @@
 @class BOXFileUploadNewVersionRequest;
 @class BOXTrashedFileRestoreRequest;
 @class BOXPreflightCheckRequest;
-@class BOXFileThumbnailRequest;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+File.h:22]: duplicated, see line 16

File: BoxContentSDK/BoxContentSDK/Requests/BOXUserAvatarRequest.m [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+- (BOXAPIOperation *)createOperation
+{
+    NSURL *URL = [self URLWithResource:BOXAPIResourceUsers
+                                    ID:self.userID
+                           subresource:BOXAPISubresourceAvatar
+                                 subID:nil];
+
+    NSMutableDictionary *queryParameters = [NSMutableDictionary dictionary];
+
+    NSString *avatarTypeString = nil;
```
Comment (crousselle) [inline BoxContentSDK/BoxContentSDK/Requests/BOXUserAvatarRequest.m:40]: minor - we could move this to be a helper to make the `createOperation` method cleaner

File: BoxContentSDK/BoxContentSDK/Requests/BOXUserAvatarRequest.m [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            dataOperation.progressBlock = ^(long long expectedTotalBytes, unsigned long long bytesReceived) {
+                [BOXDispatchHelper callCompletionBlock:^{
+                    progressBlock(bytesReceived, expectedTotalBytes);
+                } onMainThread:isMainThread];
+            };
+        }
+
+        NSOutputStream *outputStream = self.outputStream;
+        dataOperation.successBlock = ^(NSString *modelID, long long expectedTotalBytes) {
+            NSData *data = [outputStream propertyForKey:NSStreamDataWrittenToMemoryStreamKey];
```
Comment (crousselle) [inline BoxContentSDK/BoxContentSDK/Requests/BOXUserAvatarRequest.m:88]: just a question : is there any chance that any version of the avatar could be unusually large and make an output stream straight to memory an issue ?

File: BoxContentSDK/BoxContentSDK/Models/BOXUser.m [hunk 4]
```diff
@@ -114,6 +114,15 @@ - (instancetype)initWithJSON:(NSDictionary *)JSONResponse
             self.avatarURL = [NSURL URLWithString:avatarURLString];
         }
         
+        self.hasCustomAvatar = BOXAPIBooleanUnknown;
+        NSNumber *hasCustomAvatarNumber = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyHasCustomAvatar
```
Comment (crousselle) [inline BoxContentSDK/BoxContentSDK/Models/BOXUser.m:119]: is this API automatically returned by the API ? or do we have to specify all fields to get it ?

**Review by crousselle — APPROVED** [review]
Just a couple of minor questions, otherwise clean implementation :)

**Comment by crousselle** [comment]
probably have to fix tests before pushing

File: BoxContentSDK/BoxContentSDK/Requests/BOXUserAvatarRequest.m (hunk 3 above)
Comment (boxtcarpel) [reply BoxContentSDK/BoxContentSDK/Requests/BOXUserAvatarRequest.m:88]: Possibly. The information about 'large' and 'profile' types is not very clear. However this is highly unlikely right now as the Web UI prevents you from uploading a profile image that is greater than 1MB. It is safe for now to assume these are small enough (treating them the same as file thumbnails).

File: BoxContentSDK/BoxContentSDK/Models/BOXUser.m (hunk 4 above)
Comment (boxtcarpel) [reply BoxContentSDK/BoxContentSDK/Models/BOXUser.m:119]: We have to specify all fields on requesting a user.

File: BoxContentSDK/BoxContentSDK/Requests/BOXUserAvatarRequest.m (hunk 3 above)
Comment (crousselle) [reply BoxContentSDK/BoxContentSDK/Requests/BOXUserAvatarRequest.m:88]: ok cool.

**Review by crousselle — APPROVED** [review]
Awesome, +2

**Pull request merged**